tree/tree/src/TTreeCache.cxx

```diff
@@ -614,10 +630,288 @@ Int_t TTreeCache::DropBranch(const char *bname, Bool_t subbranches /*= kFALSE*/)
+      IOPos iopos = FindBranchBasket(*b);
+      if (iopos.fLen == 0) {  // Error indicator
+         continue;
+      }
+      if (iopos.fPos == pos && iopos.fLen == len) {
+         found_request = kTRUE;
+         resultBranch = b;
+         // Note that we continue to iterate; fills up the rest of the entries in the cache.
+      }
+      // At this point, we are ready to push back a new offset
+      fMissCache->fEntries.emplace_back(std::move(iopos));
+   }
+   if (R__unlikely(!found_request)) {
+      // We have gone through all the branches in this file and the requested basket
+      // doesn't appear to be in any of them.  Likely a logic error / bug.
+      fMissCache->fEntries.clear();
+   }
+   return resultBranch;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+///
+/// Process a cache miss; (pos, len) isn't in the buffer.
+///
+/// The first time we have a miss, we buffer as many baskets we can (up to the
+/// maximum size of the TTreeCache) in memory from all branches that are not in
+/// the prefetch list.
+/// 
+/// Subsequent times, we fetch all the buffers corresponding to branches that
+/// had previously seen misses.  If it turns out the (pos, len) isn't in the
+/// list of branches, we treat this as if it was the first miss.
+///
+/// Returns true if we were able to pull the data into the miss cache.
+///
+Bool_t TTreeCache::ProcessMiss(Long64_t pos, int len) {
+
+   Bool_t firstMiss = kFALSE;
+   if (fFirstMiss == -1) {
+      fFirstMiss = fEntryCurrent;
+      firstMiss = kTRUE;
+   }
+   fLastMiss = fEntryCurrent;
+   // The first time this is executed, we try to pull in as much data as we can.
+   TBranch *b = CalculateMissEntries(pos, len, firstMiss);
+   if (!b) {
+      if (!firstMiss) {
+         // TODO: this recalculates for *all* branches, throwing away the above work.
+         b = CalculateMissEntries(pos, len, kTRUE);
+      }
+      if (!b) {
+         //printf("ProcessMiss: pos %ld does not appear to correspond to a buffer in this file.\n", pos);
+         // We have gone through all the branches in this file and the requested basket
+         // doesn't appear to be in any of them.  Likely a logic error / bug.
+         fMissCache->fEntries.clear();
+         return kFALSE;
+      }
+   }
+   // TODO: this should be a set.
+   fMissCache->fBranches.push_back(b);
+
+   // OK, sort the entries
+   std::sort(fMissCache->fEntries.begin(), fMissCache->fEntries.end());
+
+   // Now, fetch the buffer.
+   std::vector<Long64_t> positions; positions.reserve(fMissCache->fEntries.size());
+   std::vector<Int_t> lengths; lengths.reserve(fMissCache->fEntries.size());
+   ULong64_t cumulative = 0;
+   for (auto &mcentry : fMissCache->fEntries) {
+         positions.push_back(mcentry.fIO.fPos);
+         lengths.push_back(mcentry.fIO.fLen);
```

Comment:
Yup.  I assume that's due to the fact this PR predates `clang-format` checks; as soon as I update, Travis-CI should spit out such fixes.